src/clr-angular/forms/common/form.ts

```diff
@@ -28,4 +52,71 @@ export class ClrForm {
   markAsTouched() {
     this.markControlService.markAsTouched();
   }
+
+  @ContentChildren(ClrLabel, { descendants: true })
+  labels: QueryList<ClrLabel>;
+
+  @HostListener('submit')
+  onFormSubmit() {
+    this.markAsTouched();
+    if (isPlatformBrowser(this.platformId)) {
+      this.invalidControls = Array.from(this.el.nativeElement.querySelectorAll('.ng-invalid'));
+      if (this.invalidControls.length > 0) {
+        this.invalidControls[0].focus();
+        this.updateAriaLive();
+      } else {
+        this.resetAriaLive();
+      }
+    }
+  }
+
+  ngOnInit() {
+    this.createAriaLiveElement();
+  }
+
+  private resetAriaLive() {
+    this.ariaLiveElement.innerHTML = '';
+  }
+
+  private updateAriaLive(): void {
+    if (this.invalidControls.length === 0) {
+      return;
+    }
+
+    this.ariaLiveText = this.commonStrings.keys.formErrorSummary;
+    this.resetAriaLive();
+
+    this.ariaLiveElement.textContent =
+      this.commonStrings.keys.formErrorSummary +
+      this.labels
+        .filter(label => label.forAttr !== undefined)
+        .filter(label => this.invalidControls.find(control => label.forAttr === control.id))
+        .map(label => label.labelText)
+        .join(' ');
+  }
+
+  private createAriaLiveElement() {
+    if (isPlatformBrowser(this.platformId)) {
+      // ClrForms need to have aria live element for screen readers.
+      this.ariaLiveElement = document.createElement('div');
+      this.renderer.setAttribute(this.ariaLiveElement, 'aria-live', 'polite');
+      this.renderer.setAttribute(this.ariaLiveElement, 'aria-live', 'polite');
+      this.renderer.setAttribute(this.ariaLiveElement, 'role', 'status');
+      this.renderer.setAttribute(this.ariaLiveElement, 'aria-atomic', 'true');
+      this.renderer.setAttribute(this.ariaLiveElement, 'aria-relevent', 'text');
+      this.renderer.addClass(this.ariaLiveElement, 'clr-sr-only');
+      this.renderer.insertBefore(this.el.nativeElement, this.ariaLiveElement, this.el.nativeElement.firstChild);
+    }
+  }
+
+  private removeAriaLiveElement() {
+    if (isPlatformBrowser(this.platformId)) {
+      this.el.nativeElement.removeChild(this.ariaLiveElement);
```

Comment:
oh we need to check if `this.ariaLiveElement` is defined besides checking platform. Also, `resetAriaLive` method needs the same check before accessing the `.innerHTML` property.

Reply:
I'll add it in the HostListener method.

Reply:
Wait why in HostListener? This method is being called in `ngOnDestroy`

Reply:
resetAriaLive is different from removeAriaLive.

It needs reset to only update the innerText so it works with screen readers. 
It needs removeAriaLive to do the cleanup wrt to removing the child from the node and deleting the ariaLiveElement. removeAriaLiveElement is only called from ngOnDestroy.

Reply:
Oh I thought you are adding it only in the HostListener. But I see you've also added that check here in this method. So looks good 👌